internal_workflow_testsuite.go

```diff
@@ -343,8 +341,10 @@ func (env *testWorkflowEnvironmentImpl) executeActivity(
 		input,
 	)
 
-	// ensure activityFn is registered to defaultTestTaskList
-	env.testSuite.RegisterActivity(activityFn)
+	if _, ok := getHostEnvironment().getActivityFn(fnName); !ok {
```

Comment:
We don't really have a use case for that private host env. As we are settling down to global registration, we should remove that private version for test suite. We could always revisit and refactor the code if we decided to do differently in the future. Offering more functionality confuse people, for example, the Registering method on TestSuite. More code also make it more likely to have more bug in it.